health/metrics/plugin_impl_metrics.go

```diff
@@ -0,0 +1,234 @@
+func (p *Plugin) Close() error {
+	if p.customProbe {
+		_, err := safeclose.CloseAll(p.HTTP)
+		return err
+	}
+
+	return nil
+}
+
+// metricsHandler handles Prometheus metrics collection.
+func (p *Plugin) metricsHandler(formatter *render.Render) http.HandlerFunc {
+	return promhttp.Handler().ServeHTTP
+}
+
+func (p *Plugin) getServiceHealth() float64 {
+	agentStatus := p.StatusCheck.GetAgentStatus()
+	// Adapt Ligato status code for now.
+	// TODO: Consolidate with that from the "Common Container Telemetry" proposal.
+	health := float64(agentStatus.State)
+	log.StandardLogger().Infof("getServiceHealth(): %f", health)
+	return health
+}
+
+// RegisterGauge registers custom gauge with specific valueFunc to report status when invoked.
+func (p *Plugin) RegisterGauge(namespace string, subsystem string, name string, help string,
+	labels prometheus.Labels, valueFunc func() float64) {
+	gaugeName := name
+	if subsystem != "" {
+		gaugeName = subsystem + "_" + gaugeName
+	}
+	if namespace != "" {
+		gaugeName = namespace + "_" + gaugeName
+	}
+	if err := prometheus.DefaultRegisterer.Register(prometheus.NewGaugeFunc(
+		prometheus.GaugeOpts{
+			// Namespace, Subsystem, and Name are components of the fully-qualified
+			// name of the Metric (created by joining these components with
+			// "_"). Only Name is mandatory, the others merely help structuring the
+			// name. Note that the fully-qualified name of the metric must be a
+			// valid Prometheus metric name.
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      name,
+
+			// Help provides information about this metric. Mandatory!
+			//
+			// Metrics with the same fully-qualified name must have the same Help
+			// string.
+			Help: help,
+
+			// ConstLabels are used to attach fixed labels to this metric. Metrics
+			// with the same fully-qualified name must have the same label names in
+			// their ConstLabels.
+			//
+			// Note that in most cases, labels have a value that varies during the
+			// lifetime of a process. Those labels are usually managed with a metric
+			// vector collector (like CounterVec, GaugeVec, UntypedVec). ConstLabels
+			// serve only special purposes. One is for the special case where the
+			// value of a label does not change during the lifetime of a process,
+			// e.g. if the revision of the running binary is put into a
+			// label. Another, more advanced purpose is if more than one Collector
+			// needs to collect Metrics with the same fully-qualified name. In that
+			// case, those Metrics must differ in the values of their
+			// ConstLabels. See the Collector examples.
+			//
+			// If the value of a label never changes (not even between binaries),
+			// that label most likely should not be a label at all (but part of the
+			// metric name).
+			ConstLabels: labels,
+		},
+		valueFunc,
+	)); err == nil {
+		log.StandardLogger().Infof("GaugeFunc('%s') registered.", gaugeName)
```

Comment:
Could you please use PluginLogDeps and then write plugin.Log.Info() ?

Reply:
sure